etc/ci_scripts/dump_artifacts.py

```diff
@@ -14,77 +17,135 @@
+        text=True,
     )
-    containers: List[str] = run_result.stdout.decode("utf-8").split("\n")
-    containers = containers[1:]  # remove the table column header
-    containers = [c for c in containers if c]  # filter empty
+    containers = result.stdout.split()
     if not containers:
-        raise ValueError(f"Couldn't find any containers for '{compose_project}'")
+        raise ValueError(f"Couldn't find any containers for '{prefix}'")
     return containers
 
 
-def _dump_docker_log(container_name: str, dir: Path) -> None:
+def _lambda_names() -> List[str]:
     """
-    run `docker logs` and dump to $DIR/$CONTAINER_NAME.log
+    Return the names of the lambdas that are running in our Localstack instance.
     """
-    destination = dir / f"{container_name}.log"
+
+    # All Localstack lambda containers begin with this prefix; what
+    # remains is the actual lambda's name.
+    #
+    # We always use the us-east-1 region, and don't change the account
+    # ID from Localstack's default of 000000000000; if that ever
+    # changes, this prefix would need to be changed accordingly.
+    lambda_container_prefix="localstack_lambda_arn_aws_lambda_us-east-1_000000000000_function_"
+    containers=_container_names_by_prefix(lambda_container_prefix)
+
+    # Chop off the prefix to get just the names of the lambda functions
+    return [name.replace(lambda_container_prefix, "") for name in containers ]
+
+
+def _dump_lambda_log(lambda_name: str, dir: Path) -> None:
+    """Dump the Cloudwatch logs of the given lambda function from Localstack.
+
+    Requires the AWS CLI to be present, and assumes a running
+    Localstack instance is available at http://localhost:4566.
+
+    These same logs are also available in the Localstack container's
+    log output, but mixed in with the logs of every other lambda. By
+    using this function, we can isolate them.
+
+    Note, however, that these logs will be only what the lambda
+    function itself outputs during its execution. There will still be
+    important information in the Localstack logs concerning how the
+    functions are *invoked* that may be useful in debugging issues.
+
+    """
+    destination = dir / f"{lambda_name}_lambda.log"
+    logging.info(f"Dumping logs for '{lambda_name}' lambda function to '{destination}'")
     with open(destination, "wb") as out_stream:
-        popen = subprocess.Popen(
+        subprocess.run(
             [
-                "docker",
+                "aws",
+                "--endpoint-url=http://localhost:4566",
                 "logs",
-                "--timestamps",
-                container_name,
+                "tail",
+                f"/aws/lambda/{lambda_name}"
             ],
             stdout=out_stream,
+            env={
+                "PATH": os.environ["PATH"],
+                # "test" is the value assumed by Localstack
+                "AWS_ACCESS_KEY_ID": "test",
+                "AWS_SECRET_ACCESS_KEY": "test"
```

Comment:
would be nice to `source` these from local-grapl.env, but not the biggest issue in the world